.claude/hooks/lint-on-save.sh

```
@@ -0,0 +1,23 @@
+FILE=$(echo "$INPUT" | python3 -c "
+import sys, json
+try:
+    d = json.load(sys.stdin)
+    path = d.get('tool_input', {}).get('file_path', '')
+    print(path)
+except:
+    print('')
+" 2>/dev/null)
+
+if [[ -n "$FILE" && "$FILE" =~ \.(ts|tsx)$ ]]; then
+  cd "$(git rev-parse --show-toplevel 2>/dev/null || pwd)"
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)